pkg/agent/client/client.go

```diff
@@ -197,30 +199,28 @@ func (c *client) FetchJWTSVID(ctx context.Context, jsr *node.JSR) (*JWTSVID, err
 	}, nil
 }
 
+// Release the underlying connection.
 func (c *client) Release() {
-	c.m.Lock()
-	defer c.m.Unlock()
-
-	if c.conn != nil {
-		c.conn.Close()
-		c.conn = nil
-	}
+	c.nodeConn.Release()
```

Comment:
we still need to set `c.nodeConn` to nil here (under the mutex) otherwise `newNodeClient` won't redial.

Reply:
+1 fixing this and adding test